### PR TITLE
add the requirements to Readme.md

### PR DESCRIPTION
**Pull request by YuZhangYuanYYZ**

### CODE REVIEW

**Comment by lanmengyujing** [comment]
looks good to me

**Review by lanmengyujing — APPROVED** [review]

File: purchasePhone.js [hunk 1]
```diff
@@ -16,18 +16,19 @@ function caculateTax(totalPrice){
...
     n = n+1;
-   totalSpendingMoney = 0;
     moneyOfAcc = n*accesorieyPrice;
     moneyOfphone = n*phonePrice;
     total = moneyOfphone+moneyOfAcc;
     totalSpendingMoney = total+caculateTax(total);   
 }
-   
-    return n;
+   function bankrupt(){
```
Comment (lanmengyujing) [inline purchasePhone.js]:
```suggestion
   function isBankrupt(thresholdSpending, totalSpendingMoney, bankMoney ){  
   return ((thresholdSpending-totalSpendingMoney)>0)&&((bankMoney-totalSpendingMoney)>0);
       
   }
```

File: purchasePhone.js [hunk 2]
```diff
@@ -16,18 +16,19 @@ function caculateTax(totalPrice){
 
 function totalSpending(){      
     var n =0;
-    while (((thresholdSpending-totalSpendingMoney)>0)&&((bankMoney-totalSpendingMoney)>0)){
+    while (bankrupt()){
```
Comment (lanmengyujing) [inline purchasePhone.js]:
```suggestion
    while (!isbankrupt(thresholdSpending, totalSpendingMoney, bankMoney )){
       ....
    }
```

File: purchasePhone.js [hunk 3]
```diff
@@ -16,18 +16,19 @@ function caculateTax(totalPrice){
 
 function totalSpending(){
```
Comment (lanmengyujing) [inline purchasePhone.js]:
```suggestion
function calculateSpending(n, accesorieyPrice, phonePrice, taxRate  ){    
   .... 
    return total;
}  
```

File: purchasePhone.js [hunk 4]
```diff
@@ -16,18 +16,19 @@ function caculateTax(totalPrice){
 
 function totalSpending(){      
     var n =0;
-    while (((thresholdSpending-totalSpendingMoney)>0)&&((bankMoney-totalSpendingMoney)>0)){
+    while (bankrupt()){
     n = n+1;
-   totalSpendingMoney = 0;
     moneyOfAcc = n*accesorieyPrice;
```
Comment (lanmengyujing) [inline purchasePhone.js]: var moneyOfAcc = n*accesorieyPrice;

File: purchasePhone.js [hunk 5]
```diff
@@ -16,18 +16,19 @@ function caculateTax(totalPrice){
```
Comment (lanmengyujing) [inline purchasePhone.js:9]:
```suggestion
function caculateTax(totalPrice, taxRate) {
  return totalPrice * taxRate;
}

```

**Review by lanmengyujing — CHANGES_REQUESTED** [review]
see comments

File: purchasePhone.js [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
-    totalSpendingMoney = total+caculateTax(total);   
-}
-   
-    return n;
-}
-phoneNumber = totalSpending();
-console.log("the number of phone you have bought is:"+phoneNumber+" the money you have spent(with tax) is:$"+totalSpendingMoney);
+    function isBankrupt(phonePrice,accesorieyPrice,taxRate,phoneAmount,thresholdSpending, bankMoney,totalSpendingMoney ){  
+    return (((thresholdSpending-totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney))>0)&&((bankMoney-totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney))>0));
+    }
```
Comment (lanmengyujing) [inline purchasePhone.js:12]:
```suggestion
 function isBankrupt(thresholdSpending, totalSpendingMoney, bankMoney) {
    return ((thresholdSpending - totalSpendingMoney) > 0) && ((bankMoney - totalSpendingMoney) > 0);
}

```

File: purchasePhone.js [hunk 7]
```diff
@@ -1,33 +1,33 @@
...
 
+    function main(){
+        var thresholdSpending =4000;
+        var bankMoney = 5000;
+        var accesorieyPrice = 10;
+        var phonePrice = 100;
+        var taxRate = 0.1;
+        var phoneAmount = 0;   
+        var totalSpendingMoney = 0; 
+        while (isBankrupt(phonePrice,accesorieyPrice,taxRate,phoneAmount,thresholdSpending,bankMoney,totalSpendingMoney )){
```
Comment (lanmengyujing) [inline purchasePhone.js]:
```suggestion
    while (!isBankrupt(threshold, total, bankMoney)) {

```

File: purchasePhone.js [hunk 8]
```diff
@@ -2,31 +2,38 @@
...
+        while (isBankrupt(thresholdSpending,bankMoney,allSpendingMoney)){
+            phoneAmount = phoneAmount+1; 
+            allSpendingMoney = spendingMoney(phonePrice,accesorieyPrice,phoneAmount,taxRate);
+            console.log("the phone number is:" +phoneAmount + "   total spending is :" +allSpendingMoney);
         }
-            totalSpendingMoney = totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney)
-            console.log("the number of phone you have bought is:"+phoneAmount+","+ "the money you have spent(with tax) is:$"+totalSpendingMoney);
+        return {
+        phoneAmount:phoneAmount,
+        allSpendingMoney: spendingMoney(phonePrice,accesorieyPrice,phoneAmount-1,taxRate),
```
Comment (lanmengyujing) [inline purchasePhone.js:26]: why calculate again here？

File: purchasePhone.js [hunk 9]
```diff
@@ -2,31 +2,38 @@
...
+    };
+    }
+    function main(){
+       
+        var thresholdSpending =4000;
+        var bankMoney = 5000; 
+        var phoneAmount = buyPhones(thresholdSpending,bankMoney).phoneAmount-1;
```
Comment (lanmengyujing) [inline purchasePhone.js:39]: buyPhones(thresholdSpending,bankMoney) is duplicate in line 34 and 35

File: purchasePhone.js [hunk 10]
```diff
@@ -2,31 +2,38 @@
...
+        var allSpendingMoney = buyPhones(thresholdSpending,bankMoney).allSpendingMoney;
+        console.log("the number of phone you have bought is:"+(phoneAmount)+","+ "the money you have spent(with tax) is:$"+allSpendingMoney);
```
Comment (lanmengyujing) [inline purchasePhone.js]: print those information in another function instead of directly call `console.log`

File: purchasePhone.js (hunk 2 above)
Comment (YuZhangYuanYYZ) [reply purchasePhone.js]: have modified

File: purchasePhone.js (hunk 9 above)
Comment (lanmengyujing) [reply purchasePhone.js:39]: here

File: purchasePhone.js (hunk 8 above)
Comment (lanmengyujing) [reply purchasePhone.js:26]: and here

File: purchasePhone.js [hunk 11]
```diff
@@ -1,33 +1,45 @@
...
+                buyPhones:"the number of phone you have bought is:"+(phoneAmount)+",",
+                spendMoney:"the money you have spent(with tax) is:$"+allSpendingMoney,
+            }
+    }
+    function main(){
+       
+        var thresholdSpending =4000;
+        var bankMoney = 5000; 
+        var phoneAmount = buyPhones(thresholdSpending,bankMoney).phoneAmount-1;
+        var allSpendingMoney = buyPhones(thresholdSpending,bankMoney).allSpendingMoney;
```
Comment (lanmengyujing) [inline purchasePhone.js:40]: here

File: purchasePhone.js [hunk 12]
```diff
@@ -1,33 +1,45 @@
...
+        var accesorieyPrice = 10;
+        var phonePrice = 100;
+        var taxRate = 0.1; 
+        while (isBankrupt(thresholdSpending,bankMoney,allSpendingMoney)){
+            phoneAmount = phoneAmount+1; 
+            allSpendingMoney = spendingMoney(phonePrice,accesorieyPrice,phoneAmount,taxRate);
+        }
+        return {
+        phoneAmount:phoneAmount,
+        allSpendingMoney: spendingMoney(phonePrice,accesorieyPrice,phoneAmount-1,taxRate),
```
Comment (lanmengyujing) [inline purchasePhone.js:26]: why calculate here again

**Pull request merged**